app/lib/import_export.rb

```diff
@@ -600,90 +332,34 @@ def extract_patients_details(patients, fields)
-    patient_details[:monitoring_status] = patient[:monitoring] ? 'Actively Monitoring' : 'Not Monitoring'
-    patient_details[:end_of_monitoring] = patient.end_of_monitoring || '' if fields.include?(:end_of_monitoring)
-    patient_details[:expected_purge_ts] = patient.expected_purge_date_exp || '' if fields.include?(:expected_purge_ts)
-    patient_details[:full_status] = patient.status&.to_s&.humanize&.downcase || '' if fields.include?(:full_status)
-    patient_details[:status] = patient.status&.to_s&.humanize&.downcase&.gsub('exposure ', '')&.gsub('isolation ', '') || '' if fields.include?(:status)
-
-    patient_details
-  end
-
-  # Gets a hash of the latest transfers of each patient
-  def get_patients_transfers(patients)
-    transfers = Transfer.latest_transfers(patients)
-    jurisdictions = Jurisdiction.find(transfers.pluck(:from_jurisdiction_id, :to_jurisdiction_id).flatten.uniq)
-    jurisdiction_paths = Hash[jurisdictions.pluck(:id, :path).map { |id, path| [id, path] }]
-    Hash[transfers.pluck(:patient_id, :created_at, :from_jurisdiction_id, :to_jurisdiction_id)
-                  .map do |patient_id, created_at, from_jurisdiction_id, to_jurisdiction_id|
-                    [patient_id, {
-                      transferred_at: created_at.rfc2822,
-                      transferred_from: jurisdiction_paths[from_jurisdiction_id],
-                      transferred_to: jurisdiction_paths[to_jurisdiction_id]
-                    }]
-                  end
-        ]
-  end
-
-  # Gets a hash of the 2 latest laboratories of each patient
-  def get_patients_laboratories(patients)
-    Laboratory.where(patient_id: patients.pluck(:id)).order(specimen_collection: :desc).group_by(&:patient_id).transform_values { |v| v.take(2) }
-  end
-
-  # Extracts assessment data values given relevant fields
+  # Extract assessment data values given relevant fields
   def extract_assessments_details(patients_identifiers, assessments, fields)
     if fields.include?(:symptoms)
-      conditions = ReportedCondition.where(assessment_id: assessments.pluck(:id))
-      symptoms = Symptom.where(condition_id: conditions&.pluck(:id)).order(:label)
-
-      conditions_hash = Hash[conditions&.pluck(:id, :assessment_id)&.map { |id, assessment_id| [id, assessment_id] }]
+      conditions_hash = Hash[assessments.joins(:reported_condition).pluck('conditions.id', :assessment_id).map { |id, assessment_id| [id, assessment_id] }]
                         .transform_values { |assessment_id| { assessment_id: assessment_id, symptoms: {} } }
-      symptoms&.each do |symptom|
-        conditions_hash[symptom[:condition_id]][:symptoms][symptom[:name]] = symptom.value
+
+      # Uses index_symptoms_on_condition_id
+      all_symptoms = Symptom.where(condition_id: conditions_hash.keys)
+      all_symptoms.where(type: 'BoolSymptom').pluck(:condition_id, :name, :bool_value).each do |(condition_id, name, bool_value)|
```

Comment:
Why is separating out each of the symptom types faster than what was there previously? Was no index being used?

Reply:
this is much faster than before because we were calling `Symptom.value` on each individual symptom which not only requires each symptom to be loaded into the ActiveRecord model (`:pluck` is almost always if not always faster), and also because `Symptom.value` calls a case/when based on the symptom type to return the value, which is becomes expensive when adding all these symptoms us.

Reply:
So I did another deep dive into this and basically what I've found is that querying for different symptom types separately saves memory (because we don't have to pluck `type` and only have to pluck that data type's `value`). However, this does slow things down due to there being multiple queries, all of which are already complicated in that they involve multiple joins, so I think it's probably best to query all of them at the same time: https://github.com/SaraAlert/SaraAlert/pull/717/commits/ab0472956fb88101b83558e527cd1558fd81a1c0#diff-c34521fcd0fdc2034d456886aae8fd35299fe908ab88d8e49c60d0b4d39ab26bR379

I left a comment in the code about my findings and this decision (happy to discuss though). I also tried an intermediate solution of plucking all the ids of those symptoms first (to hopefully avoid executing the nasty joins from the query multiple times) and then firing separate queries for each data type (to hopefully still get some memory improvements from it), but that turned out to be even slower than both existing solutions.

Reply:
So I did yet another deep dive into this and figured that maybe we can get the best of both worlds for both memory and speed by performing this query using a bit of raw SQL. I left some comments in the code to hopefully explain what I've done: https://github.com/SaraAlert/SaraAlert/blob/35627a7fc3b92ac2f4466716feff0c76d9c006f1/app/lib/import_export.rb#L345